### PR TITLE
qemu/tests/cdrom.py: Update some codes for multi host migration

### PR DESCRIPTION
**Pull request by spcui**
Update serial number for Multihost()

Signed-off-by: Shuping Cui <scui@redhat.com>

### CODE REVIEW

**Comment by spcui** [comment]
ID: 1288975 1288976

File: qemu/tests/cdrom.py [hunk 1]
```diff
@@ -385,7 +386,7 @@ def get_match_cdrom(session, serial_num):
             raise error.TestFail("Could not find the corresponding cdrom"
                                  "in guest which is same in Qemu CML.")
 
-    def get_testing_cdrom_device(session, cdrom_dev_list, serial_num=None):
+    def get_testing_cdrom_device(vm, cdrom_dev_list, serial_num=None):
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: Why use vm here instead of session directly?

Comment (spcui) [inline qemu/tests/cdrom.py]: check Line:373, will use vm, so update it.

Comment (CongSmile) [inline qemu/tests/cdrom.py]: cool~, thanks.

**Comment by luckyh** [comment]
Acked-by: @luckyh

File: qemu/tests/cdrom.py [hunk 2]
```diff
@@ -349,7 +349,7 @@ def get_device(vm, dev_file_path):
                 raise error.TestFail("Could not find a valid cdrom device")
         return device
 
-    def get_match_cdrom(session, serial_num):
+    def get_match_cdrom(vm, serial_num):
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: For this func(), can use the avocado-framework/avocado-vt#324 directly now.

File: qemu/tests/cdrom.py [hunk 3]
```diff
@@ -398,7 +362,8 @@ def get_testing_cdrom_device(session, cdrom_dev_list, serial_num=None):
                 cdrom_dev_list.remove(winutil_drive)
                 testing_cdrom_device = cdrom_dev_list[-1]
             else:
-                testing_cdrom_device = get_match_cdrom(session, serial_num)
+                testing_cdrom_device = utils_misc.get_linux_drive_path(
+                    session, serial_num)
```
Comment (CongSmile) [inline qemu/tests/cdrom.py]: For the 2nd parameter, it can be a wwn, which requires int type.
but the parameter which is transferred is a str.

**Comment by suqinhuang** [comment]
LGTM

Acked-by: Suqin Huang

File: qemu/tests/cdrom.py [hunk 4]
```diff
@@ -762,14 +762,15 @@ def test(self):
             self.is_src = params.get("hostid") == self.srchost
             self.mig = self.mig_type(test, params, env, False, )
             self.cdrom_size = int(params.get("cdrom_size", 10))
+            cdrom = params.get("cdroms", "").split()[-1]
```
Comment (xutian) [inline qemu/tests/cdrom.py]: params.objects("cdroms")[-1]

File: qemu/tests/cfg/multi_host.cfg [hunk 5]
```diff
@@ -15,7 +15,7 @@
             vms = "vm1"
             start_vm = no
             migration_test_command = help
-            migration_bg_command = "cd /tmp; nohup tcpdump -q -i any -t ip host localhost &"
+            migration_bg_command = "cd /tmp; nohup tcpdump -q -i any -t ip host localhost > tcpdump.out 2>&1 &"
```
Comment (xutian) [inline qemu/tests/cfg/multi_host.cfg]: why not  "nohup tcpdump -q -i any -t ip host localhost 2>/tmp/tcpdump.out>&1 &"

File: qemu/tests/cfg/multi_host.cfg [hunk 6]
```diff
@@ -250,9 +250,28 @@
...
                 - cdrom_test:
                     only Linux
-                    only mig_online.after_login_vm
                     type = cdrom
-                    cdrom_cd1 = orig.iso
+                    preprocess_env = yes
+                    cdrom_cd1 = "/mnt/nfs_images/orig.iso"
+                    target_cdrom = cdrom_cd1
+                    drive_serial_cd1 = test_abcd123
+                    drive_werror_cd1 = stop
```
Comment (xutian) [inline qemu/tests/cfg/multi_host.cfg]: It looks 'werror' not works for IDE interface .

File: qemu/tests/cfg/multi_host.cfg [hunk 7]
```diff
@@ -250,9 +250,28 @@
 
                 - cdrom_test:
                     only Linux
-                    only mig_online.after_login_vm
                     type = cdrom
-                    cdrom_cd1 = orig.iso
+                    preprocess_env = yes
+                    cdrom_cd1 = "/mnt/nfs_images/orig.iso"
```
Comment (xutian) [inline qemu/tests/cfg/multi_host.cfg]: does this change impact local cdrom test ?

File: qemu/tests/cfg/multi_host.cfg (hunk 5 above)
Comment (spcui) [inline qemu/tests/cfg/multi_host.cfg]: orig cmd will create ignoring input message, like this
# cd /tmp; nohup tcpdump -q -i any -t ip host localhost &
[1] 28085
# nohup: ignoring input and appending output to ‘nohup.out’
and cannot get the correct output and status, so update it.

File: qemu/tests/cfg/multi_host.cfg (hunk 7 above)
Comment (spcui) [inline qemu/tests/cfg/multi_host.cfg]: no, this config only used by multi_host.

File: qemu/tests/cfg/multi_host.cfg (hunk 6 above)
Comment (spcui) [inline qemu/tests/cfg/multi_host.cfg]: tested it, work well for ide.

**Pull request merged**